src/ccache.cpp

```diff
@@ -931,9 +939,64 @@ to_cache(Context& ctx,
     return nonstd::make_unexpected(Statistic::missing_cache_file);
   }
 
+  // MSVC compiler always print the input file name to stdout,
+  // plus parts of the warnings/error messages.
+  // So we have to fusion that into stderr...
+  // Transform \r\n into \n. This way ninja won't produce empty newlines
+  // for the /showIncludes argument.
```

Comment:
> Could you expand on this a bit? What happens if `\r\n` are kept?

I think ccache from msvc build is generally broken here. 
I got part of the system tests working again on a private branch when ccache is built with mingw.
But when it is built with msvc, already the first tests fail because in the cached output (preprocessor, stdout, ...) there are always two line breaks instead of one.

Reply:
`/showInlcudes` is used by ninja to do its own dependency tracking. Without the conversion to `\n` ninja would insert newlines in the compilation output and would look like a mess.

With the new code to replace `\r\n` with `\n` also a MSVC build of ccache works as expected. The previous code relied on iostreams which are a bit broken on MSVC.

I tested with Qt Creator:
1. build with no cache, save the output of ninja
2. build with ccache, save the ouptut of ninja

Compare the two, warnings and other messages that the compiler produced were the same.

Reply:
Actually I used `> build.log 2>&1`, which is not the same as having stdout and stderr separate.

Reply:
The extra newlines come because STDERR_FILENO on Windows is open in text mode, which transforms newlines. Handled without a hack, I mean with a less of a hack, in #954 .

Reply:
Commits b8c81b760935cf84c69b520ee266cbea22ce8ec3 and 60a6d2d51bcd6fb6e618c1be7aba933ae08d0a06 from #954  work very well. 
I tested them on top of #943 and was able to remove the hack in the run script.

I would suggest to cherry-pick them before integrating.

Reply:
I'm intentionally keeping #954 only as fixes for this, so no need to cherry-pick anything, it can go in right after this one at any time.